lib/logstash/inputs/elasticsearch.rb

```diff
@@ -475,4 +483,31 @@ def validate_value(value, validator)
     end
   end
   extend(PositiveWholeNumberValidator)
+
+  module FieldReferenceValidator
```

Comment:
I think this needs to be extracted into a separate support mixin, since it will be helpful to use in other plugins implementing `target`.

It would be great if we could provide this in Logstash core, but the way the config mixin works currently is non-extensible, and anything we add to it cannot be consumed by plugins without preventing those plugins from working on older versions of Logstash.